src/components/bar/utils/monitors/index.ts

```diff
@@ -12,24 +19,80 @@ import { JSXElement } from 'src/core/types';
  * @returns BarLayout configuration for the specified monitor, falling back to default if not found
  */
 export const getLayoutForMonitor = (monitor: number, layouts: BarLayouts): BarLayout => {
-    const matchingKey = Object.keys(layouts).find((key) => key === monitor.toString());
-    const wildcard = Object.keys(layouts).find((key) => key === '*');
+    const [rootKey, rootLayout] = _getResolveLayoutForMonitor(monitor, layouts);
 
-    if (matchingKey !== undefined) {
-        return layouts[matchingKey];
-    }
+    let left = rootLayout.left;
+    let middle = rootLayout.middle;
+    let right = rootLayout.right;
 
-    if (wildcard) {
-        return layouts[wildcard];
+    let layout = rootLayout;
+    const visited = new Array([rootKey]);
+    while (
+        layout.extends !== undefined &&
+        (left === undefined || middle === undefined || right.length === undefined)
```

Comment:
```suggestion
        (left === undefined || middle === undefined || right === undefined)
```
I imagine the `.length` was accidental?

Reply:
It was a leftover of a test, nice catch, ty!